Fix AssertionError for docker RepositoryTestCase

- tests:  tests/foreman/cli/test_repository.py::RepositoryTestCase::test_negative_synchronize_docker_repo_with_mix_valid_invalid_tags    tests/foreman/cli/test_repository.py::RepositoryTestCase::test_positive_synchronize_docker_repo_set_tags_later  tests/foreman/cli/test_repository.py::RepositoryTestCase::test_positive_synchronize_docker_repo_with_tags_whitelist

- Error:
self = <tests.foreman.cli.test_repository.RepositoryTestCase testMethod=test_positive_synchronize_docker_repo_set_tags_later>

    @run_only_on('sat')
    @tier2
    def test_positive_synchronize_docker_repo_set_tags_later(self):
        """Verify that adding tags whitelist and re-syncing after
        synchronizing full repository doesn't remove content that was
        already pulled in
    
        :id: 97f2087f-6041-4242-8b7c-be53c68f46ff
    
        :expectedresults: Non-whitelisted tags are not removed
        """
        tags = 'latest'
        repo = self._make_repository({
            'content-type': 'docker',
            'docker-upstream-name': 'hello-world',
            'url': DOCKER_REGISTRY_HUB,
        })
        Repository.synchronize({'id': repo['id']})
        repo = Repository.info({'id': repo['id']})
        # Field is displayed only if there is any content
        with self.assertRaises(KeyError):
            repo['container-image-tags-filter']
>       self.assertGreaterEqual(int(repo['content-counts']['container-image-tags']), 2)
E       AssertionError: 0 not greater than or equal to 2



self = <tests.foreman.cli.test_repository.RepositoryTestCase testMethod=test_negative_synchronize_docker_repo_with_mix_valid_invalid_tags>

    @run_only_on('sat')
    @tier2
    def test_negative_synchronize_docker_repo_with_mix_valid_invalid_tags(self):
        """Set tags whitelist to contain both valid and invalid (non-existing)
        tags. Check if only whitelisted tags are synchronized
    
        :id: 75668da8-cc94-4d39-ade1-d3ef91edc812
    
        :expectedresults: Only whitelisted tag is synchronized
        """
        tags = ['latest', gen_string('alpha')]
        repo = self._make_repository({
            'content-type': 'docker',
            'docker-upstream-name': 'alpine',
            'url': DOCKER_REGISTRY_HUB,
            'docker-tags-whitelist': ",".join(tags),
        })
        Repository.synchronize({'id': repo['id']})
        repo = Repository.info({'id': repo['id']})
        [self.assertIn(tag, repo['container-image-tags-filter']) for tag in tags]
>       self.assertEqual(int(repo['content-counts']['container-image-tags']), 1)
E       AssertionError: 0 != 1

tests/foreman/cli/test_repository.py:1070: AssertionError

self = <tests.foreman.cli.test_repository.RepositoryTestCase testMethod=test_positive_synchronize_docker_repo_with_tags_whitelist>

    @run_only_on('sat')
    @tier2
    @upgrade
    def test_positive_synchronize_docker_repo_with_tags_whitelist(self):
        """Check if only whitelisted tags are synchronized
    
        :id: aa820c65-2de1-4b32-8890-98bd8b4320dc
    
        :expectedresults: Only whitelisted tag is synchronized
        """
        tags = 'latest'
        repo = self._make_repository({
            'content-type': 'docker',
            'docker-upstream-name': 'alpine',
            'url': DOCKER_REGISTRY_HUB,
            'docker-tags-whitelist': tags,
        })
        Repository.synchronize({'id': repo['id']})
        repo = Repository.info({'id': repo['id']})
        self.assertIn(tags, repo['container-image-tags-filter'])
>       self.assertEqual(int(repo['content-counts']['container-image-tags']), 1)
E       AssertionError: 0 != 1

tests/foreman/cli/test_repository.py:1015: AssertionError

--- a/tests/foreman/cli/test_repository.py
+++ b/tests/foreman/cli/test_repository.py
@@ -15,6 +15,7 @@
 
 :Upstream: No
 """
+from wait_for import wait_for
 
 from fauxfactory import gen_alphanumeric, gen_string
 from robottelo import ssh
@@ -123,6 +124,21 @@ class RepositoryTestCase(CLITestCase):
             options[u'product-id'] = self.product['id']
 
         return make_repository(options)
+
+    def _get_image_tags_count(self, repo=None):
+        repo_detail = Repository.info({'id': repo['id']})
+        return repo_detail
+
+    def _validated_image_tags_count(self, repo=None):
+        if bz_bug_is_open(1664631):
+            wait_for(
+                lambda: int(self._get_image_tags_count(repo=repo)
+                            ['content-counts']['container-image-tags']) > 0,
+                timeout=30,
+                delay=2,
+                logger=self.logger
+            )
+        return self._get_image_tags_count(repo=repo)
 
     @tier1
     @upgrade
@@ -1010,7 +1026,7 @@ class RepositoryTestCase(CLITestCase):
             'docker-tags-whitelist': tags,
         })
         Repository.synchronize({'id': repo['id']})
-        repo = Repository.info({'id': repo['id']})
+        repo = self._validated_image_tags_count(repo=repo)
         self.assertIn(tags, repo['container-image-tags-filter'])
         self.assertEqual(int(repo['content-counts']['container-image-tags']), 1)
 
@@ -1032,20 +1048,21 @@ class RepositoryTestCase(CLITestCase):
             'url': DOCKER_REGISTRY_HUB,
         })
         Repository.synchronize({'id': repo['id']})
-        repo = Repository.info({'id': repo['id']})
+        repo = self._validated_image_tags_count(repo=repo)
         # Field is displayed only if there is any content
         with self.assertRaises(KeyError):
             repo['container-image-tags-filter']
-        self.assertGreaterEqual(int(repo['content-counts']['container-image-tags']), 2)
-
+        self.assertGreaterEqual(int(repo['content-counts']
+                                    ['container-image-tags']), 2)
         Repository.update({
             'id': repo['id'],
             'docker-tags-whitelist': tags,
         })
         Repository.synchronize({'id': repo['id']})
-        repo = Repository.info({'id': repo['id']})
+        repo = self._validated_image_tags_count(repo=repo)
         self.assertIn(tags, repo['container-image-tags-filter'])
-        self.assertGreaterEqual(int(repo['content-counts']['container-image-tags']), 2)
+        self.assertGreaterEqual(int(repo['content-counts']
+                                    ['container-image-tags']), 2)
 
     @run_only_on('sat')
     @tier2
@@ -1065,7 +1082,7 @@ class RepositoryTestCase(CLITestCase):
             'docker-tags-whitelist': ",".join(tags),
         })
         Repository.synchronize({'id': repo['id']})
-        repo = Repository.info({'id': repo['id']})
+        repo = self._validated_image_tags_count(repo=repo)
         [self.assertIn(tag, repo['container-image-tags-filter']) for tag in tags]
         self.assertEqual(int(repo['content-counts']['container-image-tags']), 1)
 

